src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs

```diff
@@ -1193,20 +1240,30 @@ protected override void EndProcessing()
             {
                 if (ParameterSetName.Equals(InvokeCommandCommand.InProcParameterSet))
                 {
-                    if (_steppablePipeline != null)
-                    {
-                        _steppablePipeline.End();
+                    try
+                    {   
+                        if (_steppablePipeline != null)
+                        {
+                            _steppablePipeline.End();
+                        }
+                        else
+                        {
+                            ScriptBlock.InvokeUsingCmdlet(
+                                contextCmdlet: this,
+                                useLocalScope: !NoNewScope,
+                                errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                                dollarUnder: AutomationNull.Value,
+                                input: _input,
+                                scriptThis: AutomationNull.Value,
+                                args: ArgumentList);
+                        }
                     }
-                    else
+                    finally
                     {
-                        ScriptBlock.InvokeUsingCmdlet(
-                            contextCmdlet: this,
-                            useLocalScope: !NoNewScope,
-                            errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
-                            dollarUnder: AutomationNull.Value,
-                            input: _input,
-                            scriptThis: AutomationNull.Value,
-                            args: ArgumentList);
+                        if (MyInvocation.BoundParameters.ContainsKey(nameof(StrictModeVersion)))
```

Comment:
How about `if (_savedStrictModeVersion is not null)`?

Reply:
We cannot check for null since it is a valid value.